packages/x-components/src/composables/use-state.ts

```diff
@@ -0,0 +1,22 @@
+import { computed } from 'vue';
+import { ExtractState, XModuleName } from '../x-modules/x-modules.types';
+import { useStore } from './use-store';
+
+/**
+ * Generates a computed property which returns the selected state.
+ *
+ * @param module - The {@link XModuleName} of the getter.
+ * @param path - The state path.
+ * @returns The state properties of the module.
+ * @public
+ */
+export function useState<Module extends XModuleName, Path extends keyof ExtractState<Module>>(
+  module: Module,
+  path: Path
+): ExtractState<Module>[Path] {
+  const $store = useStore();
+  const state = computed((): ExtractState<Module>[Path] => {
+    return $store.state.x[module][path];
+  });
+  return state.value;
```

Comment:
Returning the `value` instead of the computed property breaks the reactivity. Better to return the computed property directly.

```suggestion
  return computed(() => $store.state.x[module][path]);
```

Reply:
Thanks!!